cdap-app-fabric/src/test/java/co/cask/cdap/ConditionalWorkflowApp.java

```diff
@@ -186,29 +161,7 @@ public void beforeSubmit(MapReduceContext context) throws Exception {
       WordCount.configureJob((Job) context.getHadoopJob(), inputPath, outputPath);
       // Put something in the token
       context.getWorkflowToken().put("action_type", "MapReduce");
-      context.getWorkflowToken().put("start_time", String.valueOf(System.currentTimeMillis()));
-    }
-
-    // TODO: CDAP-2894 MapReduce counters should be available implicitly from the WorkflowToken.
-    // We yet to figure out the scheme.
-    @Override
-    public void onFinish(boolean succeeded, MapReduceContext context) throws Exception {
-      if (!succeeded) {
-        return;
-      }
-
-      WorkflowToken workflowToken = context.getWorkflowToken();
-      if (workflowToken == null) {
-        throw new IllegalStateException("WorkflowToken cannot be null when the MapReduce program" +
-                                          " is started from the Workflow.");
-      }
-
-      Counters counters = ((Job) context.getHadoopJob()).getCounters();
-      for (CounterGroup group : counters) {
-        for (Counter counter : group) {
-          workflowToken.put(group.getName() + "." + counter.getName(), String.valueOf(counter.getValue()));
-        }
-      }
+      context.getWorkflowToken().put("start_time", Value.of(System.currentTimeMillis()));
```

Comment:
Can you call it "start.time" instead? I know it's just a unit-test, however someone will eventually copy the concept and make it into example, which using "." looks much nicer and is the convention in hadoop and cdap.